core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -382,29 +409,33 @@ export class RteHelperModalComponent {
     } else {
       for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
         const caName = this.tmpCustomizationArgs[i].name;
-        if (caName === 'video_id') {
-          const temp = this.tmpCustomizationArgs[i].value;
-          customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
-            temp.toString()
-          );
-        } else if (caName === 'text' && this.currentRteIsLinkEditor) {
-          // Set the link `text` to the link `url` if the `text` is empty.
-          (
-            customizationArgsDict as {
-              [Prop in CustomizationArgsNameAndValueArray[number]['name']]: CustomizationArgsNameAndValueArray[number]['value'];
-            }
-          )[caName] =
-            this.tmpCustomizationArgs[i].value ||
-            this.tmpCustomizationArgs[i - 1].value;
-        } else {
-          (
-            customizationArgsDict as {
-              [Prop in CustomizationArgsNameAndValueArray[number]['name']]: CustomizationArgsNameAndValueArray[number]['value'];
-            }
-          )[caName] = this.tmpCustomizationArgs[i].value;
+        if (this.componentId === this.COMPONENT_ID_VIDEO) {
+          if (caName === 'video_id') {
+            this.tmpCustomizationArgs[i].value =
+              this.extractVideoIdFromVideoUrl(
+                this.tmpCustomizationArgs[i].value.toString()
+              );
+          }
+        } else if (this.componentId === this.COMPONENT_ID_LINK) {
+          if (caName === 'text') {
+            // Set the link `text` to the link `url` if the `text` is empty.
+            (
+              customizationArgsDict as {
+                [Prop in CustomizationArgsNameAndValueArray[number]['name']]: CustomizationArgsNameAndValueArray[number]['value'];
+              }
+            )[caName] =
+              this.tmpCustomizationArgs[i].value ||
+              this.tmpCustomizationArgs[i - 1].value;
+          }
         }
+        (
+          customizationArgsDict as {
+            [Prop in CustomizationArgsNameAndValueArray[number]['name']]: CustomizationArgsNameAndValueArray[number]['value'];
+          }
+        )[caName] = this.tmpCustomizationArgs[i].value;
```

Comment:
Wouldn't this have the effect of overwriting the 'text' customization arg in the link case that was only just set in lines 426-428?

Reply:
The specific condition in lines 426-428  guarantees that the 'text' customization argument is never empty when it's overwritten by the general assignment. Therefore, the overwriting doesn't cause any issues since it's just set to its current value being the actuall text if not empty or the url if empty.

Reply:
Hi @joanapeixinho, sorry for the late reply, I was extremely busy over the weekend.

I'm not sure I understand still, sorry. Suppose this.tmpCustomizationArgs[i - 1].value is nonempty and this.tmpCustomizationArgs[i].value is empty. Then lines 426-428 set the value of the arg to `this.tmpCustomizationArgs[i - 1].value`. But then this line, which runs after it, will set it back to the empty `this.tmpCustomizationArgs[i].value`. Right?

(Also if this analysis is correct, and the code is indeed wrong, then please write a test to catch this case in the future. Thanks!)

Reply:
@seanlip Done!